sodiumpp/include/sodiumpp/locked_string.h

```diff
@@ -22,13 +22,13 @@ class locked_string final
              * for move construction for std::string use static methods:
              * move_from_locked_string() and move_from_not_locked_string
              */
-            explicit locked_string(std::string && str) noexcept; ///< only move, not call mlock()
+            //explicit locked_string(std::string && str) noexcept; ///< only move, not call mlock()
     public:
             locked_string() = default;
             locked_string(const locked_string &);
-            locked_string(locked_string &&) = default;
+            locked_string(locked_string &&other);
```

Comment:
we need special op to avoid SSO.

Perhaps we should after all move to std::vector for this one. It's the one case where in fact std::vector behavior does differ from std::string, we need guarantee here that memory address of data before/after moving (std::move) will be the same and no optimization tricks can change it.

@da2ce7 :trollface: :godmode: